docs/topics/opt-in-requirements.md

```diff
@@ -271,21 +393,56 @@ fun getTime(): Time {}
 ```
 
 Note that for some language elements, an opt-in requirement annotation is not applicable:
-* You cannot annotate a backing field or a getter of a property, just the property itself.
-* You cannot annotate a local variable or a value parameter.
+
+* You can't annotate a backing field or a getter of a property, just the property itself.
+* You can't annotate a local variable or a value parameter.
+
+## Require opt-in to extend API
+
+There may be times when you want more granular control over which specific parts of your API can be used and
+extended. For example, when you have some API that is stable to use but:
+
+* **Unstable to implement** due to ongoing evolution, such as when you have a family of interfaces where you expect to add new abstract functions without default implementations.
+* **Delicate or fragile to implement**, such as individual functions that need to behave in a coordinated manner.
+* **Has a contract that may be weakened in the future** in a backwards-incompatible manner for external implementations, such as changing an input parameter `T` to a nullable version `T?` where the code didn't previously consider `null` values.
+
+In such cases, you want to require users to opt in to your API before they can extend it further.
+For example, by inheriting from the API or declaring implementations for abstract functions. Using the [`@SubclassOptInRequired`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-subclass-opt-in-required/)
```

Comment:
Hmm 🤔 I'm a bit confused by this sentence: For example, by inheriting from the API or declaring implementations for abstract functions. 
what do you think about having this together with the previous sentence/ somehow rewording to connect it more? 
....extend it further. Extending the API includes, inheriting from the API or declaring implementations...

Reply:
What about:

> In such cases, you can require users to opt in to your API before they can extend it further. Users can extend your API by inheriting from the API or implementing abstract functions. By using the `@SubclassOptInRequired` annotation, you can enforce this requirement to opt-in for open or abstract classes and non-functional interfaces.

Reply:
nice one! 👍  I like it!